src/ColorPicker/Halogen/Layout.purs

```diff
@@ -0,0 +1,459 @@
+  , view: \{color, value, onValueInput, onBlur} → pure $
+      renderInput
+        { root: classes.root
+        , label: classes.label
+        , prefix: "#"
+        , child: HH.input $
+            [ HP.type_ HP.InputText
+            , HP.classes
+              $  classes.elem
+              <> (guard (isInvalid value) *> (classes.elemInvalid))
+            , HP.title "Hex"
+            , HP.value $ maybe' (\_ → toString color) _.value value
+            , HP.placeholder "Hex"
+            , HE.onValueInput $ onValueInput >>> Just
+            , HE.onBlur $ onBlur >>> Just
+            ]
+        }
+  }
+  where
+  toString =  \{color} → String.toUpper $ String.drop 1 $ Color.toHexString color
+
+
+isValid ∷ Maybe InputTextValue → Boolean
+isValid = maybe true _.isValid
+
+isInvalid ∷ Maybe InputTextValue → Boolean
+isInvalid = not isValid
+
+renderInput ∷ ∀ i p.
+  { child ∷ HH.HTML i p
+  , prefix ∷ String
+  , label ∷ Classes
+  , root ∷ Classes
+  }
+  → HH.HTML i p
+renderInput {root, label, prefix, child} =
+  HH.label
+    [ HP.classes root]
+    [ HH.span [HP.classes label] [HH.text prefix]
+    , child
+    ]
+
+
+-- Internal helpers
+
+
+confRed ∷ PreNumConf
+confRed =
+  { title: "Red"
+  , placeholder: "R"
+  , prefix: "R"
+  , range: MinMax 0.0 256.0
+  }
+
+confGreen ∷ PreNumConf
+confGreen =
+  { title: "Green"
+  , placeholder: "G"
+  , prefix: "G"
+  , range: MinMax 0.0 256.0
+  }
+
+confBlue ∷ PreNumConf
+confBlue =
+  { title: "Blue"
+  , placeholder: "B"
+  , prefix: "B"
+  , range: MinMax 0.0 256.0
+  }
+
+confHue ∷ PreNumConf
+confHue =
+  { title: "Hue"
+  , placeholder: "H"
+  , prefix: "H"
+  , range: MinMax 0.0 360.0
+  }
+confSaturation ∷ PreNumConf
+confSaturation =
+  { title: "Saturation"
+  , placeholder: "S"
+  , prefix: "S"
+  , range: MinMax 0.0 100.0
+  }
+confLightness ∷ PreNumConf
+confLightness =
+  { title: "Lightness"
+  , placeholder: "L"
+  , prefix: "L"
+  , range: MinMax 0.0 100.0
+  }
+
+confValue ∷ PreNumConf
+confValue =
+  { title: "Value"
+  , placeholder: "V"
+  , prefix: "V"
+  , range: MinMax 0.0 100.0
+  }
+
+
+hasValRound ∷ Num.HasNumberInputValue Number
+hasValRound = Num.numberHasNumberInputValue
+  {fromString = Num.numberHasNumberInputValue.fromString >>> map roundFractionalNum}
+
+hasValCail ∷ Num.HasNumberInputValue Number
+hasValCail = Num.numberHasNumberInputValue
+  {fromString = Num.numberHasNumberInputValue.fromString >>> map roundNum}
+
+roundFractionalNum ∷ Number → Number
+roundFractionalNum n = roundNum (n * scalar) / scalar
+  where
+  scalar = 100.0
+
+asInt ∷ Number → Int
+asInt = floor
+
+roundNum ∷ Number → Number
+roundNum = round
+
+type RecordHSLA = { h ∷ Number, s ∷ Number, l ∷ Number, a ∷ Number }
+type RecordHSVA = { h ∷ Number, s ∷ Number, v ∷ Number, a ∷ Number }
+type RecordRGBA = { r ∷ Int, g ∷ Int, b ∷ Int, a ∷ Number }
+
+modifyHSL ∷ (RecordHSLA → RecordHSLA) → Dynamic Color
+modifyHSL f { hsl } = let {h, s, l, a} = f (force hsl) in Color.hsla h s l a
```

Comment:
What these function do? To me having `Color` and not having `LazyColor` makes this functions look like this 
```purescript
modifyHSL f = hsla <<< f <<< Color.toHSLA
  where 
  hsla {h, s, l, a } = Color.hsla h s l a
```